eox_nelp/openedx_filters/xapi/filters.py

```diff
@@ -80,19 +85,23 @@ def run_filter(self, transformer, result):  # pylint: disable=arguments-differ,
 
         Arguments:
             transformer <XApiTransformer>: Transformer instance.
-            result <Activity>: Object activity for events related to enrollments.
+            result <Activity>: Course object activity for any kind of event
 
         Returns:
             Activity: Modified activity.
         """
         course_id = extract_course_id_from_string(result.id)
 
-        if course_id:
+        if course_id and result.definition.type == constants.XAPI_ACTIVITY_COURSE:
             # Get course course data
             course = get_course_from_id(course_id)
             display_name = course["display_name"]
             description = course["short_description"]
-            course_language = course["language"] or DEFAULT_LANGUAGE  # Set default value if language is not found
+            course_language = course.get("language")
+
+            # Set default value if language is not found
+            if not course_language or course_language == constants.EN:
+                course_language = DEFAULT_LANGUAGE
```

Comment:
What is DEFAULT_LANGUAGE?
https://github.com/nelc/event-routing-backends/blob/master/event_routing_backends/processors/xapi/constants.py

Reply:
https://github.com/eduNEXT/eox-nelp/blob/master/eox_nelp/processors/xapi/constants.py#L6